apisix/discovery/consul/init.lua

```diff
@@ -246,7 +270,7 @@ function _M.connect(premature, consul_server, retry_delay)
                 -- add services to table
                 local nodes = up_services[service_name]
                 for  _, node in ipairs(result.body) do
-                    local svc_address, svc_port = node.ServiceAddress, node.ServicePort
+                    local svc_address, svc_port = node.Service.Address, node.Service.Port
```

Comment:
need to check whether `node.Service` is nil

Reply:
checked done